alicloud/resource_alicloud_cs_kubernetes_autoscaler.go

```diff
@@ -0,0 +1,552 @@
+		}
+
+		err = DeployAutoscaler(args, fmt.Sprintf(defaultAutoscalerImage, regionId), kubeConfPath)
+
+		if err != nil {
+			return WrapError(fmt.Errorf("failed to deploy autoscaler,because of %v", err))
+		}
+	}
+	return nil
+}
+
+// delete autoscaler deployment
+func resourceAlicloudCSKubernetesAutoscalerDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	regionId := client.RegionId
+
+	clusterId := d.Get("cluster_id").(string)
+
+	if clusterId == "" {
+		return WrapError(fmt.Errorf("please provide the cluster_id in region %s", regionId))
+	}
+
+	kubeConfPath, err := DownloadUserKubeConf(client, clusterId)
+
+	if err != nil {
+		return WrapError(fmt.Errorf("failed to download kubeconf from cluster,because of %v", err))
+	}
+
+	return DeleteAutoscaler(kubeConfPath)
+}
+
+// update scaling group config
+func UpdateScalingGroupConfiguration(client *connectivity.AliyunClient, groupId, userData string, labels string, taints string) (err error) {
+
+	describeScalingConfigurationsRequest := ess.CreateDescribeScalingConfigurationsRequest()
+	describeScalingConfigurationsResponse, err := client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+		describeScalingConfigurationsRequest.RegionId = client.RegionId
+		describeScalingConfigurationsRequest.ScalingGroupId = groupId
+		return essClient.DescribeScalingConfigurations(describeScalingConfigurationsRequest)
+	})
+
+	configurations, ok := describeScalingConfigurationsResponse.(*ess.DescribeScalingConfigurationsResponse)
+
+	if ok != true {
+		return WrapError(fmt.Errorf("failed to parse DescribeScalingConfigurationsResponse of %s", groupId))
+	}
+
+	if err != nil {
+		return WrapErrorf(err, "failed to invoke CreateDescribeScalingConfigurations,because of %v", err)
+	}
+
+	addDebug("DescribeScalingConfigurations", configurations, describeScalingConfigurationsRequest)
+
+	if len(configurations.ScalingConfigurations.ScalingConfiguration) == 0 {
+		//todo  create configuration
+		return WrapError(fmt.Errorf("please create the default scaling configuration of group %s", groupId))
+	} else {
+		defaultConfiguration := configurations.ScalingConfigurations.ScalingConfiguration[0]
+		// modify the default one
+		modifyScalingConfigurationRequest := ess.CreateModifyScalingConfigurationRequest()
+		modifyScalingConfigurationResponse, err := client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+			modifyScalingConfigurationRequest.RegionId = client.RegionId
+			modifyScalingConfigurationRequest.UserData = userData
+			modifyScalingConfigurationRequest.Tags = createScalingGroupTags(labels, taints)
+			modifyScalingConfigurationRequest.ScalingConfigurationId = defaultConfiguration.ScalingConfigurationId
+			return essClient.ModifyScalingConfiguration(modifyScalingConfigurationRequest)
+		})
+
+		if err != nil {
+			return WrapErrorf(err, "failed to invoke ModifyScalingConfiguration,because of %v", err)
```

Comment:
Please change line 311 to `	return WrapErrorf(err, DefaultErrorMsg, defaultConfiguration.ScalingConfigurationId, modifyScalingConfigurationRequest.GetActionName(), AlibabaCloudSdkGoERROR)
)
`